### PR TITLE
Move Pipeline Failure Notifications to Separate Workflow

### PR DESCRIPTION
**Pull request by kierramarie**
### Description

The notifications to create an issue and a slack notification have been moved to their own github actions workflow. The Semantic release pipeline has been updated to use this workflow.

### Release required?

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: .github/workflows/create-gh-issue-and-slack-message.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+      - name: Create Github and Slack notifications
+        run: |
+          response=$(curl -sL \
+              -X POST \
+              -H "Accept: application/vnd.github.v3+json" \
+              -H "Authorization: Bearer ${{ secrets.GH_TOKEN }}" \
+              -H "X-GitHub-Api-Version: 2022-11-28" \
+              https://api.github.com/repos/${{ github.repository }}/issues \
+              -d '{"title":"Release Pipeline failed","body":"For details see: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}","labels":["ci-failure"]}')
```
Comment (daniel-butler-irl) [inline .github/workflows/create-gh-issue-and-slack-message.yml:24]: This is looking good, but I think we need inputs for the title, maybe labels too. You can set defaults for the values with what you have hardcoded here.

File: .github/workflows/create-gh-issue-and-slack-message.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_call:
+
+jobs:
+  Create-Notifications:
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v3
```
Comment (daniel-butler-irl) [inline .github/workflows/create-gh-issue-and-slack-message.yml:14]: I do not think you need to checkout the code in this job

Comment (kierramarie) [reply .github/workflows/create-gh-issue-and-slack-message.yml:25]: It is needed to create the git issue (and then the slack message).

Comment (kierramarie) [reply .github/workflows/create-gh-issue-and-slack-message.yml:25]: Unless we want to hard code that in.

**Comment by kierramarie** [comment]
@daniel-butler-irl is there anything else you'd like me to do for this PR?

**Comment by ocofaigh** [comment]
@daniel-butler-irl want to do a final review here?

**Comment by daniel-butler-irl** [comment]
Approved once it has been manually validated

**Review by daniel-butler-irl — APPROVED** [review]

**Pull request merged**